src/overlays/actors/ovl_En_Bb/z_en_bb.c

```diff
@@ -32,65 +105,1137 @@ const ActorInit En_Bb_InitVars = {
+    }
+    this->collider.list[0].dim.worldSphere.radius =
+        this->collider.list[0].dim.modelSphere.radius * this->collider.list[0].dim.scale;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809B8E34.s")
+void EnBb_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnBb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809B8EF4.s")
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809B96EC.s")
+void EnBb_SetupFlameTrail(EnBb* this) {
+    this->action = BB_FLAME_TRAIL;
+    this->moveMode = BBMOVE_NOCLIP;
+    this->actor.flags &= ~1;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    EnBb_SetupAction(this, EnBb_FlameTrail);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809B977C.s")
+void EnBb_FlameTrail(EnBb* this, GlobalContext* globalCtx) {
+    if (this->actor.params == ENBB_KILL_TRAIL) {
+        if (this->actor.parent == NULL) {
+            EnBb_SetupKill(this, globalCtx);
+        }
+    } else {
+        if (this->timer == 0) {
+            if (((EnBb*)this->targetActor)->flameScaleY != 0.0f) {
+                Math_SmoothScaleMaxMinF(&this->flameScaleY, this->actor.scale.y, 1.0f, this->actor.scale.y * 0.1f,
+                                        0.0f);
+                Math_SmoothScaleMaxMinF(&this->flameScaleX, this->actor.scale.z, 1.0f, this->actor.scale.z * 0.1f,
+                                        0.0f);
+                if (this->flamePrimAlpha != this->trailMaxAlpha) {
+                    this->flamePrimAlpha += 10;
+                    if (this->trailMaxAlpha < this->flamePrimAlpha) {
+                        this->flamePrimAlpha = this->trailMaxAlpha;
+                    }
+                }
+            } else {
+                if (!this->flamePrimAlpha) {
+                    Actor_Kill(&this->actor);
+                    return;
+                } else if (this->flamePrimAlpha <= 20) {
+                    this->flamePrimAlpha = 0;
+                } else {
+                    this->flamePrimAlpha -= 20;
+                }
+            }
+            this->actor.posRot.pos = this->actor.parent->pos4;
+        } else {
+            this->timer--;
+            this->actor.shape.rot.y = this->actor.posRot.rot.y = this->actor.parent->posRot.rot.y;
+        }
+    }
+    if (this->actor.parent != NULL) {
+        this->actor.velocity.y = this->actor.parent->velocity.y;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809B9A54.s")
+void EnBb_SetupKill(EnBb* this, GlobalContext* globalCtx) {
+    if (this->actor.params <= ENBB_BLUE) {
+        this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+        this->actor.speedXZ = -7.0f;
+        this->timer = 5;
+        this->actor.shape.rot.x += 0x4E20;
```

Comment:
have you considered: 
```suggestion
        this->actor.shape.rot.x += 20000;
```
I know some people prefer hex angles and some prefer whatever looks like nice round numbers, I'm just checking with you out of curiosity